addons/assigngear/functions/fn_assignGear.sqf

```diff
@@ -76,81 +99,108 @@ _unit setUnitLoadout (configFile >> 'EmptyLoadout');
                 if !(_weapon isEqualTo '') then {_unit addWeapon _weapon};
             };
-            case 19: { // secondaryAttachments
+            case IDX_SECONDARY_ATTACHMENTS: {
                 {_unit addSecondaryWeaponItem _x} forEach _x;
             };
-            case 20: { // sidearmweapon
+            case IDX_SIDEARM_WEAPON: {
                 private _weapon = selectRandom _x;
                 if !(_weapon isEqualTo '') then {_unit addWeapon _weapon};
             };
-            case 21: { // sidearmattachments
+            case IDX_SIDEARM_ATTACHMENTS: {
                 {_unit addHandgunItem _x} forEach _x;
             };
-            case 22: { // Unit traits
+            case IDX_TRAITS: {
                 {
                     [_unit, _x] call FUNC(setUnitTrait);
                 } forEach _x;
             };
-            case 23: { // code
+            case IDX_PRIMARY_MAGAZINE: {
+                private _magazine = selectRandom _x;
+                if (toLower _magazine != 'default') then {
+                    // Sadly I don't think this can't be done faster, primaryMagazine should then only be used when really needed.
+                    private _weapon = primaryWeapon _unit;
+                    private _weaponMags = [_weapon, false] call CBA_fnc_compatibleMagazines;
+                    private _backup = (primaryWeaponMagazine _unit) select {_x in _weaponMags};  // Save what game already placed inside the weapon, so it can be put back in inventory
+                    _unit addPrimaryWeaponItem _magazine;
+                    [_backup] call _fnc_addMagazines;
+                };
+            };
+            case IDX_PRIMARY_GRENADE: {
+                private _magazine = selectRandom _x;
+                if (toLower _magazine != 'default') then {
+                    // Sadly I don't think this can't be done faster, primaryGrenade should then only be used when really needed.
+                    private _weapon = primaryWeapon _unit;
+                    private _weaponMags = [_weapon, false] call CBA_fnc_compatibleMagazines;
+                    private _weaponGrenades = ([_weapon, true] call CBA_fnc_compatibleMagazines) - _weaponMags;
+                    private _backup = (primaryWeaponMagazine _unit) select {_x in _weaponGrenades};
+                    _unit addPrimaryWeaponItem _magazine;
+                    [_backup] call _fnc_addMagazines;
+                };
+            };
+            case IDX_SECONDARY_MAGAZINE: {
+                private _magazine = selectRandom _x;
+                if (toLower _magazine != 'default') then {
+                    private _backup = secondaryWeaponMagazine _unit;
+                    _unit addSecondaryWeaponItem _magazine;
+                    [_backup] call _fnc_addMagazines;
+                };
+            };
+            case IDX_SIDEARM_MAGAZINE: {
+                private _magazine = selectRandom _x;
+                if (toLower _magazine != 'default') then {
```

Comment:
`!=` and `==` are case-insensitive

For case sensitive comparison, you use `isEqualTo`

Reply:
well then, reverting this one change